packages/core/src/controllers/verify.ts

```diff
@@ -1,68 +1,140 @@
-  // this means Verify API is not enabled for the current projectId and there's no point in trying to initialize it again.
-  private verifyDisabled = false;
-
-  constructor(public projectId: string, public logger: Logger) {
-    super(projectId, logger);
+  private verifyUrlV2 = VERIFY_SERVER_V2;
+  private publicKey?: jwk;
+
+  constructor(public projectId: string, public logger: Logger, public store: IKeyValueStorage) {
+    super(projectId, logger, store);
     this.logger = generateChildLogger(logger, this.name);
-    this.verifyUrl = VERIFY_SERVER;
     this.abortController = new AbortController();
     this.isDevEnv = isNode() && process.env.IS_VITEST;
+    console.log("Verify v2 init", this.verifyUrlV2);
+    this.init();
   }
 
-  public init: IVerify["init"] = async (params) => {
-    if (this.verifyDisabled) return;
-
-    // ignore on non browser environments
-    if (isReactNative() || !isBrowser()) return;
-
-    const verifyUrl = this.getVerifyUrl(params?.verifyUrl);
-    // if init is called again with a different url, remove the iframe and start over
-    if (this.verifyUrl !== verifyUrl) {
-      this.removeIframe();
-    }
-    this.verifyUrl = verifyUrl;
+  get storeKey(): string {
+    return `verify:public:key`;
+  }
 
-    try {
-      await this.createIframe();
-    } catch (error) {
-      this.logger.info(`Verify iframe failed to load: ${this.verifyUrl}`);
-      this.logger.info(error);
-      // if the iframe fails to load, disable verify
-      this.verifyDisabled = true;
+  public init = async () => {
+    this.publicKey = await this.store.getItem(this.storeKey);
+    console.log("persistedKey", this.publicKey);
+    if (this.publicKey && toMiliseconds(this.publicKey?.expiresAt) < Date.now()) {
+      console.log("public key expired");
+      await this.removePublicKey();
     }
+    if (this.publicKey) return;
+    const key = await this.fetchPublicKey();
+    console.log("public key", key);
+    await this.persistPublicKey(key);
   };
 
   public register: IVerify["register"] = async (params) => {
-    if (!this.initialized) {
-      this.addToQueue(params.attestationId);
-      await this.init();
-    } else {
-      this.sendPost(params.attestationId);
+    console.log("register", params);
+    const { id, decryptedId } = params;
+    const url = `${this.verifyUrlV2}/attestation?projectId=${this.projectId}`;
+    let src = "";
+    try {
+      const response = await fetch(url, {
+        method: "POST",
+        body: JSON.stringify({ id, decryptedId }),
+        headers: {
+          origin: "https://8951-78-130-198-143.ngrok-free.app/",
+        },
+      });
+      const { srcdoc } = await response.json();
+      src = srcdoc;
+    } catch (e) {
+      console.error("error", e);
+      return;
     }
+    console.log("srcdoc", src);
+
+    const abortTimeout = this.startAbortTimer(ONE_SECOND * 2);
+    const attestatiatonJwt = await new Promise((resolve) => {
+      const abortListener = () => {
```

Comment:
Is this perhaps a simpler way to do this vs all these add and remove listener calls? https://x.com/mattpocockuk/status/1817958766756733278